lib/OpenQA/VcsProvider.pm

```diff
@@ -0,0 +1,36 @@
+# Copyright 2030 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::VcsProvider;
+
+use Mojo::Base -base, -signatures;
+use Mojo::JSON qw(encode_json);
+use Mojo::URL;
+
+has 'app';
+
+sub report_status_to_github ($self, $statuses_url, $params, $scheduled_product_id = undef, $callback = undef) {
+    $params->{context} //= 'openqa';
+    $params->{description} //= 'openQA test run';
+    $params->{target_url} //= "WIP/admin/productlog?id=$scheduled_product_id" if defined $scheduled_product_id;
```

Comment:
This is still WIP. Since we want to show this URL on an external site we cannot just use the `url_for` helper.